lottie/src/main/java/com/airbnb/lottie/network/DefaultLottieFetchResult.java

```diff
@@ -0,0 +1,70 @@
+package com.airbnb.lottie.network;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class DefaultLottieFetchResult implements LottieFetchResult {
+
+  @NonNull
+  private final HttpURLConnection connection;
+
+  public DefaultLottieFetchResult(@NonNull HttpURLConnection connection) {
+    this.connection = connection;
+  }
+
+  @Override public boolean isSuccessful() {
+    try {
+      return resultCode() / 100 == 2;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  @Override public int resultCode() throws IOException {
+    return connection.getResponseCode();
+  }
+
+  @Nullable @Override public InputStream bodyByteStream() throws IOException {
```

Comment:
Under what situation would this return null?

Reply:
marked it NonNull